fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -935,7 +934,10 @@ def cast_buffers_(
     # if buffers are already on the right device and/or dtype this is just python loop cost
     for key, buf in module.named_buffers(recurse=False):
```

Comment:
let's assert dtype is either fp16 or fp32? perhaps also update the docstring to state that this function is only switching dtype for fp buffers?